mmeval/metrics/bleu.py

```diff
@@ -0,0 +1,193 @@
+        ] for r in references]
+        predictions_token: Sequence[Sequence[str]] = [
+            tokenizer_fn(line) if line else [] for line in predictions
+        ]
+        pred_len = 0
+        references_len = 0
+        for prediction, references in zip(predictions_token, references_token):
+            pred_len = len(prediction)
+            references_len_list = [len(reference) for reference in references]
+            references_len_diff = [
+                abs(len(prediction) - length) for length in references_len_list
+            ]
+            # In the multi sentence reference, the one whose length is closest
+            # to the predicted sentence is selected to record the length.
+            min_index = references_len_diff.index(min(references_len_diff))
+            references_len = references_len_list[min_index]
+
+            pred_counter: Counter = get_n_gram(prediction, self.n_gram)
+            reference_counter: Counter = Counter()
+            for reference in references:
+                # Take intersection for the n_gram of references.
+                reference_counter |= get_n_gram(reference, self.n_gram)
+            # Union the n_gram of prediction and references.
+            counter_clip = pred_counter & reference_counter
+            precision_matches = np.zeros(self.n_gram)
+            precision_total = np.zeros(self.n_gram)
+            for counter in counter_clip:
+                precision_matches[len(counter) - 1] += counter_clip[counter]
+            for counter in pred_counter:
+                precision_total[len(counter) - 1] += pred_counter[counter]
+            result = (pred_len, references_len, precision_matches,
+                      precision_total)
+            self._results.append(result)
+
+    def compute_metric(
+            self, results: List[Tuple[int, int, np.ndarray,
+                                      np.ndarray]]) -> dict:
+        """Compute the bleu metric.
+
+        This method would be invoked in ``BaseMetric.compute`` after
+        distributed synchronization.
+
+        Args:
+            results (List[Tuple[int, int, np.ndarray, np.ndarray]]):
+                A list that consisting the tuple of correct numbers.
+                Tuple contains pred_len, references_len,
+                precision_matches, precision_total.
+                This list has already been synced across all ranks.
+
+        Returns:
+            Dict[str, float]: The computed bleu score.
+        """
+        pred_len = 0
+        references_len = 0
+        precision_matches = np.zeros(self.n_gram)
+        precision_total = np.zeros(self.n_gram)
+        for result in results:
+            pred_len += result[0]
+            references_len += result[1]
+            precision_matches += result[2]
+            precision_total += result[3]
+
+        if min(precision_matches) == 0.0:
+            return {'bleu': np.array(0.0)}
+        if self.smooth:
```

Comment:
Suggesting add a blank line here~
```suggestion
        if min(precision_matches) == 0.0:
            return {'bleu': np.array(0.0)}

        if self.smooth:
```